keras/saving/hdf5_format.py

```diff
@@ -615,6 +615,27 @@ def load_optimizer_weights_from_hdf5_group(hdf5_group):
   return [weights_group[weight_name] for weight_name in optimizer_weight_names]
 
 
+def save_model_weights_to_hdf5_group(f, model):
+  """Save top-level weights of a model to a HDF5 group.
+
+  Args:
+      f: HDF5 group.
+      model: model instance.
+  """
+  save_weights_to_hdf5_group(f, model.layers)
+  weights = _legacy_weights(model)
+  weight_values = backend.batch_get_value(weights)
+  weight_names = [w.name.encode('utf8') for w in weights]
+  save_attributes_to_hdf5_group(f, 'weight_names', weight_names)
+  for name, val in zip(weight_names, weight_values):
+    param_dset = f.create_dataset(name, val.shape, dtype=val.dtype)
+    if not val.shape:
+      # scalar
+      param_dset[()] = val
+    else:
+      param_dset[:] = val
+
+
 def save_weights_to_hdf5_group(f, layers):
```

Comment:
Thinking about this more, it seems we should go with the approach of:

- Keeping as much as possible of the current code
- Just adding support for saving/loading top-level weights (`model._trainable_weights` and `model._non_trainable_weights`)

We can do this by refactoring `def save_weights_to_hdf5_group(f, layers):` into the following:

```python
def save_weights_to_hdf5_group(f, model):
   layers = model.layers
   ...  # Same logic as before
   ...  # Add logic for saving top-level weights to a new group
```

The following block will need to be abstracted to a new function (same for loading), since we need to reuse it twice:

```python
    weight_values = backend.batch_get_value(weights)
    weight_names = [w.name.encode('utf8') for w in weights]
    save_attributes_to_hdf5_group(g, 'weight_names', weight_names)
    for name, val in zip(weight_names, weight_values):
      param_dset = g.create_dataset(name, val.shape, dtype=val.dtype)
      if not val.shape:
        # scalar
        param_dset[()] = val
      else:
        param_dset[:] = val
```
What do you think?